crates/analytics/src/connector_events/core.rs

```diff
@@ -0,0 +1,27 @@
+use api_models::analytics::connector_events::ConnectorEventsRequest;
+use common_utils::errors::ReportSwitchExt;
+use error_stack::{IntoReport, ResultExt};
+
+use super::events::{get_connector_events, ConnectorEventsResult};
+use crate::{errors::AnalyticsResult, types::FiltersError, AnalyticsProvider};
+
+pub async fn connector_events_core(
+    pool: &AnalyticsProvider,
+    req: ConnectorEventsRequest,
+    merchant_id: String,
+) -> AnalyticsResult<Vec<ConnectorEventsResult>> {
+    let data = match pool {
+        AnalyticsProvider::Sqlx(_) => Err(FiltersError::NotImplemented(
+            "Connector Events not implemented for SQLX",
+        ))
+        .into_report()
+        .attach_printable("SQL Analytics is not implemented for Connector Events"),
+        AnalyticsProvider::Clickhouse(pool) => get_connector_events(&merchant_id, req, pool).await,
+        AnalyticsProvider::CombinedSqlx(_sqlx_pool, ckh_pool)
+        | AnalyticsProvider::CombinedCkh(_sqlx_pool, ckh_pool) => {
+            get_connector_events(&merchant_id, req, ckh_pool).await
+        }
```

Comment:
```suggestion
        AnalyticsProvider::Clickhouse(pool)
        | AnalyticsProvider::CombinedSqlx(_sqlx_pool, ckh_pool)
        | AnalyticsProvider::CombinedCkh(_sqlx_pool, ckh_pool) => {
            get_connector_events(&merchant_id, req, ckh_pool).await
        }
```

Reply:
`  --> crates/analytics/src/connector_events/core.rs:19:9
   |
19 |         AnalyticsProvider::Clickhouse(ckh_pool)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ pattern doesn't bind `_sqlx_pool`
20 |         | AnalyticsProvider::CombinedSqlx(_sqlx_pool, ckh_pool)
   |                                           ---------- variable not in all patterns
21 |         | AnalyticsProvider::CombinedCkh(_sqlx_pool, ckh_pool) => {
   |                                          ---------- variable not in all patterns`